toolchain/sem_ir/formatter.cpp

```diff
@@ -299,13 +291,22 @@ auto Formatter::FormatScopeIfUsed(InstNamer::ScopeId scope_id,
   llvm::SaveAndRestore scope(scope_, scope_id);
   // Note, we don't use OpenBrace() / CloseBrace() here because we always want
   // a newline to avoid misformatting if the first instruction is omitted.
-  out_ << inst_namer_.GetScopeName(scope_id) << " {\n";
+  out_ << "\n" << inst_namer_.GetScopeName(scope_id) << " {\n";
   indent_ += 2;
   for (const InstId inst_id : block) {
-    TentativeOutputScope scope(*this, tentative_inst_chunks_[inst_id.index]);
-    FormatInst(inst_id);
+    // Unlike code blocks, we don't note when a scope elide entries, because
+    // they're less likely to be related to the elided code.
+    if (ShouldFormatInst(inst_id)) {
+      if (is_tentative) {
+        TentativeOutputScope scope(*this,
+                                   tentative_inst_chunks_[inst_id.index]);
+        FormatInst(inst_id);
+      } else {
+        FormatInst(inst_id);
+      }
+    }
```

Comment:
I had trouble understanding this comment because I assumed it was explaining the next line or the next block, but I couldn't figure out how they related. Since it's explaining the absence of some code, how about putting it where that code would go?

```suggestion
    if (ShouldFormatInst(inst_id)) {
      if (is_tentative) {
        TentativeOutputScope scope(*this,
                                   tentative_inst_chunks_[inst_id.index]);
        FormatInst(inst_id);
      } else {
        FormatInst(inst_id);
      }
    }
    // Unlike code blocks, we don't note when a scope elide entries, because
    // they're less likely to be related to the elided code.
```

You could even put it inside an empty `else` block, but that may be too cute.

Reply:
I've rephrased the comment to try to make it fit better, can you take another look?

I don't want to put it after, as suggested:

- The suggested control flow implies the comment unconditionally applies, even though the comment is actually conditional.
- You're right that an `else` block would solve that, but I would prefer to avoid comment-only control flow: for me, even dead control flow makes it harder to skim code.